crates/oxc_cli/src/lib.rs

```diff
@@ -61,6 +61,17 @@ impl Cli {
             })
             .sum();
 
+        // https://eslint.org/docs/latest/use/command-line-interface#--max-warnings
+        if self.cli_options.max_warnings > -1 {
+            let max_warnings = self.cli_options.max_warnings as usize;
+            if number_of_diagnostics > max_warnings {
+                return CliRunResult::MaxWarningsExceeded {
+                    number_of_warnings: number_of_diagnostics,
+                    max_warnings,
+                };
+            }
+        }
```

Comment:
It seems this option is used to control the total number of diagnostics to print? Which means you'll have to keep a count and let it stop printing once the count has exceeded.

Maybe this'll help, just maybe:
https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.scan

Reply:
I couldn't figure out how to use `scan` so converted it to [regular loops](https://github.com/Boshen/oxc/pull/77/commits/e9ccb55e167aa721022f813ab25da1b1fcdb4b3e#diff-70497152b3a021a31c5a3aa54940c396cbad87a81f9d6a6fc24e56e78a0583a7R44-R69). It feels _more_ readable to me, but may not be as performant, so happy for you or someone else to rewrite in a way that fits best 😄

Reply:
I think this does take a perf hit, iirc par_iter() runs on multiple threads

Reply:
I assumed as much, according to `rayon` docs:

> The `ParallelIterator` module makes it easy to write parallel programs using an iterator-style interface.

If somebody wants to help me achieve the same outcome as what I've written using `par_iter` instead, that would be super helpful 🙏🏼

Reply:
Ill try to take another stab at it today if I have some time 🙂 Did you try the scan method that Boshen linked above?

Reply:
I tried reading the docs but couldn't bring myself to interpret it today 😅